lib/ansible/modules/database/mongodb/mongodb_info.py

```diff
@@ -0,0 +1,444 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2020, Andrew Klychkov (@Andersson007) <aaklychkov@mail.ru>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: mongodb_info
+
+short_description: Gather information about MongoDB instance.
+
+description:
+- Gather information about MongoDB instance.
+
+author: Andrew Klychkov (@Andersson007)
+
+version_added: '2.10'
+
+options:
+  filter:
+    description:
+    - Limit the collected information by comma separated string or YAML list.
+    - Allowable values are C(general), C(databases), C(total_size), C(parameters), C(users), C(roles).
+    - By default, collects all subsets.
+    - You can use '!' before value (for example, C(!users)) to exclude it from the information.
+    - If you pass including and excluding values to the filter, for example, I(filter=!general,users),
+      the excluding values, C(!general) in this case, will be ignored.
+    required: no
+    type: list
+    elements: str
+  login_user:
+    description:
+    - The MongoDB user to login with.
+    - Required when I(login_password) is specified.
+    required: no
+    type: str
+  login_password:
+    description:
+    - The password used to authenticate with.
+    - Required when I(login_user) is specified.
+    required: no
+    type: str
+  login_database:
+    description:
+    - The database where login credentials are stored.
+    required: no
+    type: str
+    default: 'admin'
+  login_host:
```

Comment:
I guess it's time to add a documentation fragment for MongoDB :) But probably it's better to wait until the modules have been moved before starting with that.

Reply:
fully agree. It crossed my mind to. And, in addition, i'm going to create something like ```mongo.py``` for the general part of arg_spec and other code that could be shared among all the modules.